File: backend/src/main/java/com/baniyamart/backend/service/OrderService.java
package com.baniyamart.backend.service;

import java.util.List;
import java.util.Optional;

import com.baniyamart.backend.model.Order;

public interface OrderService {

    Order createOrder(Order order);

    List<Order> getAllOrders();

    Optional<Order> getOrderById(String id);

    List<Order> getOrdersByUserId(String userId);

    Order updateOrderStatus(String id, String status); // We'll validate enum in impl

    void deleteOrder(String id);

}

File: frontend/src/services/orderService.js
import axios from "axios";

const API_BASE_URL = "http://localhost:8080/api/orders";

const orderService = {
  // Create an Order\
  createOrder: async (orderData) => {
    const response = await axios.post(API_BASE_URL, orderData);
    return response.data;
  },

  // Get All Orders

  getAllOrders: async () => {
    const response = await axios.get(API_BASE_URL);
    return response.data;
  },

  // Get Order by ID
  getOrderById: async (id) => {
    const response = await axios.get(`${API_BASE_URL}/${id}`);
    return response.data;
  },

  // Get Orders by User ID
  getOrdersByUserId: async (userId) => {
    const response = await axios.get(`${API_BASE_URL}/user/${userId}`);
    return response.data;
  },

  // Update Order Status
  updateOrderStatus: async (id, status) => {
    const response = await axios.put(`${API_BASE_URL}/${id}/status`, {
      status,
    });
    return response.data;
  },

  // Delete Order
  deleteOrder: async (id) => {
    const response = await axios.delete(`${API_BASE_URL}/${id}`);
    return response.data;
  },
};

export default orderService;
